modules/thermal_hydraulics/src/components/Component2D.C

```diff
@@ -353,47 +353,82 @@ Component2D::buildMesh()
   }
 
   // Create boundary IDs and associated boundary names
-  _inner_bc_id.push_back(mesh().getNextBoundaryId());
-  _outer_bc_id.push_back(mesh().getNextBoundaryId());
+  _inner_bc_id = mesh().getNextBoundaryId();
+  _outer_bc_id = mesh().getNextBoundaryId();
   _boundary_name_inner = genName(name(), "inner");
   _boundary_name_outer = genName(name(), "outer");
+  _boundary_name_to_area[_boundary_name_inner] = computeRadialBoundaryArea(_length, 0.0);
+  _boundary_name_to_area[_boundary_name_outer] =
+      computeRadialBoundaryArea(_length, getTotalWidth());
   if (_n_sections > 1 && _axial_region_names.size() == _n_sections)
     for (unsigned int i = 0; i < _n_sections; i++)
     {
       _axial_inner_bc_id.push_back(mesh().getNextBoundaryId());
       _axial_outer_bc_id.push_back(mesh().getNextBoundaryId());
-      _boundary_names_axial_inner.push_back(genName(name(), _axial_region_names[i], "inner"));
-      _boundary_names_axial_outer.push_back(genName(name(), _axial_region_names[i], "outer"));
+      const BoundaryName boundary_name_axial_inner =
+          genName(name(), _axial_region_names[i], "inner");
+      const BoundaryName boundary_name_axial_outer =
+          genName(name(), _axial_region_names[i], "outer");
+      _boundary_names_axial_inner.push_back(boundary_name_axial_inner);
+      _boundary_names_axial_outer.push_back(boundary_name_axial_outer);
+      _boundary_name_to_area[boundary_name_axial_inner] =
+          computeRadialBoundaryArea(_lengths[i], 0.0);
+      _boundary_name_to_area[boundary_name_axial_outer] =
+          computeRadialBoundaryArea(_lengths[i], getTotalWidth());
     }
 
   // exterior axial boundaries
-  _start_bc_id.push_back(mesh().getNextBoundaryId());
-  _end_bc_id.push_back(mesh().getNextBoundaryId());
+  _start_bc_id = mesh().getNextBoundaryId();
+  _end_bc_id = mesh().getNextBoundaryId();
   _boundary_name_start = genName(name(), "start");
   _boundary_name_end = genName(name(), "end");
+  _boundary_name_to_area[_boundary_name_start] = computeAxialBoundaryArea(0.0, getTotalWidth());
+  _boundary_name_to_area[_boundary_name_end] = computeAxialBoundaryArea(0.0, getTotalWidth());
   if (_names.size() > 1)
+  {
+    Real y1 = 0.0;
```

Comment:
For cylindrical HS, you should initialize y1 to `inner_radius`?

Reply:
No, `inner_radius` is accounted for in the area integral functions; `y1` is just the actual y coordinate.